File: t/java/Sm/TestClass.java

package Sm;

public class TestClass
{
    private TestClassContext _fsm;

    public TestClass()
    {
        _fsm = new TestClassContext(this);

        // Uncomment to see debug output.
        // _fsm.setDebugFlag(true);
    }

    public statemap.FSMContext getFSM() {
        return _fsm;
    }

    public void NoArg() {
        System.out.println("No arg");
    }

    public void Output(String str) {
        System.out.println(str);
    }

    public void Output_n(String str1, int n, String str2) {
        System.out.print(str1);
        System.out.print(n);
        System.out.println(str2);
    }

    public boolean isOk() {
        return true;
    }

    public boolean isNok() {
        return false;
    }

    public void Evt_1() {
        _fsm.Evt_1();
    }

    public void Evt_2() {
        _fsm.Evt_2();
    }

    public void Evt_3() {
        _fsm.Evt_3();
    }

    public void Evt1(int n) {
        _fsm.Evt1(n);
    }

    public void Evt2(int n) {
        _fsm.Evt2(n);
    }

    public void Evt3(int n) {
        _fsm.Evt3(n);
    }

}

File: t/javascript/Sm/TestClass.js

require('./../../../runtime/javascript/statemap.js');
require('./TestClassContext.js');

function TestClass () {
    this.fsm = new TestClassContext(this);

    // Uncomment to see debug output.
    //this.fsm.setDebugFlag(true);
}
global.TestClass = TestClass;

TestClass.prototype.NoArg = function () {
    process.stdout.write("No arg\n");
}

TestClass.prototype.Output = function (str) {
    process.stdout.write(str + "\n");
}

TestClass.prototype.Output_n = function (str1, n, str2) {
    process.stdout.write(str1);
    process.stdout.write(n.toString());
    process.stdout.write(str2);
    process.stdout.write("\n");
}

TestClass.prototype.isOk = function () {
    return true;
}

TestClass.prototype.isNok = function () {
    return false;
}

TestClass.prototype.Evt_1 = function () {
    this.fsm.Evt_1();
}

TestClass.prototype.Evt_2 = function () {
    this.fsm.Evt_2();
}

TestClass.prototype.Evt_3 = function () {
    this.fsm.Evt_3();
}

TestClass.prototype.Evt1 = function (n) {
    this.fsm.Evt1(n);
}

TestClass.prototype.Evt2 = function (n) {
    this.fsm.Evt2(n);
}

TestClass.prototype.Evt3 = function (n) {
    this.fsm.Evt3(n);
}

